### ISSUE TITLE
Unstable training on d3il m1

### ISSUE THREAD

**Issue by NathanWalt** [irom-princeton/dppo#27]
Thank you for sharing your excellent work. I've been testing your code on the d3il task with the cfg file provided in the repository. Yet finetuning is highly unstable, with the reward falling to around zero and trajectory completely deviating from the original one after about 7 rounds of sampling and updating. Is such phenomenon caused by poor hyperparameter selection? Or is it simply hard for DPPO to learn in tasks such as d3il?
<img width="398" alt="image" src="https://github.com/user-attachments/assets/253d270b-472f-4f44-b15d-6acfc42a66c3" />

**Comment by allenzren** [comment]
Hmm that is unusual. Could you share the exact config and the task?

Also, just to check, did you install my own fork of D3IL?

**Comment by allenzren** [comment]
We did tune the hyperparameters for other tasks since some updates to our formulation after the initial release. But we didn’t check the D3IL tasks again. I can take a look

**Comment by NathanWalt** [comment]
I follow the installation/install_d3il.md to install the d3il environment. For the config I use the cfg/d3il/finetune/avoid_m1/ft_ppo_diffusion_mlp.yaml file in your repository and use the checkpoint downloaded from the url in this yaml file. Could you kindly share your hyperparameter configuration?

**Comment by allenzren** [comment]
Yep I am trying bigger batch size now. Will let you know

**Comment by allenzren** [comment]
I updated the configs https://github.com/irom-princeton/dppo/commit/e7f73dffc131570ef7129d5ed1bc98a05cf030ab. Basically the batch size had to be multiplied by the number of fine-tuned denoising steps, thus the effective batch size matches the original ones used in the paper.
![Screenshot 2024-12-24 at 1 36 53 AM](https://github.com/user-attachments/assets/a0a00750-21ba-4c2a-8292-201fb4a27d34)

Thanks for pointing this out too! We should've updated the D3IL configs when we introduced the new gradient update (https://github.com/irom-princeton/dppo/releases/tag/v0.6).

**Comment by NathanWalt** [comment]
Thank you for your update! I'll try it out myself later.
